queries.sql

```diff
@@ -21,3 +21,102 @@ SELECT * FROM animals WHERE name != 'Gabumon';
 
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+
+-- Operation 1
+BEGIN;
+-- Update species to 'unspecified'
+UPDATE animals
+SET species = 'unspecified';
+-- Verify the change
+SELECT * FROM animals;
+-- Rollback the change
+ROLLBACK;
+-- Verify species column reverted to previous state
+SELECT * FROM animals;
+
+
+
+-- Operation 2
+BEGIN;
+-- Update species to 'digimon' for names ending in 'mon'
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+-- Update species to 'pokemon' for animals without a species
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+-- Verify the changes
+SELECT * FROM animals;
+-- Commit the transaction
+COMMIT;
+-- Verify changes persist after commit
+SELECT * FROM animals;
+
+
+
+-- Operation 3
+-- Start a transaction
+BEGIN;
+
+-- Delete all records in the animals table
+DELETE FROM animals;
+
+-- Rollback the transaction
+ROLLBACK;
+
+-- Check if records still exist after rollback
+SELECT * FROM animals;
```

Comment:
- [ ] As a good practice, let's ensure that changes are verified before and after rollback, and our code should be similar to:
```sql
  BEGIN;

   DELETE FROM animals WHERE ;
   SELECT COUNT(*) FROM ANIMALS;

   ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
```

Reply:
Good suggestion 👍 
Implemented.